link-grammar/prune.c

```diff
@@ -573,14 +573,14 @@ static inline bool matches_S(connector_table *ct, Connector * c, char dir)
  */
 static int mark_dead_connectors(connector_table *ct, Exp * e, char dir)
 {
-	int count;
-	count = 0;
+	int count = 0;
+	Connector dummy;
+	init_connector(&dummy);
```

Comment:
At least gcc5.1.1 (that I use) is very smart. However, init_connector() is a function in another file and it doesn't know, by default, to inline it, so this pull request makes it inline.

BTW, **gcc**, for some time now (gcc4.8) knows to do link-time optimizations (-flto) , including global inlining. I tested it and it produces a library which is much smaller (I don't know yet why, I would expect it to be bigger due to the inlining).
However, when I tested it I couldn't find evidences for an improved speed - the variation between runs was too high for a meaningful comparisons by a few runs. But at that time I didn't use yet my current method of averaging 200 runs, so I will eventually repeat this test.

I will resend to the same branch (but after you incorporate the length-limit pull request, which I have just also resent to the same branch - in order  to prevent a possible conflict again).